v3/kafka.go

```diff
@@ -402,28 +474,31 @@ func (client *KafkaClient) publishAndRetryFailure(context context.Context, topic
 	config := client.publishConfig
 	topic = constructTopic(client.prefix, topic)
 
-	go func() {
-		err := backoff.RetryNotify(func() error {
-			return client.publishEvent(context, topic, eventName, config, message)
-		}, backoff.WithMaxRetries(newPublishBackoff(), maxBackOffCount),
-			func(err error, _ time.Duration) {
+	if !config.Async {
+		go func() {
```

Comment:
could we remove the entire code inside `if !config.Async` (I mean the goroutine func)? I don't think it makes sense anymore since it can only be used if `!config.Async`, while putting it inside goroutine makes it called asynchronously

Reply:
It will keep backward compatible here. So example we have some user that has config.Async is false and still calling the `Publish` that expect the call not blocking.